client_resolver.go

```diff
@@ -34,26 +41,135 @@ func (c *Client) watchAddressUpdates(r Resolver) {
 		case <-r.Done():
 			return
 		case addrs := <-r.UpdateChan():
-			c.attemptConnection(addrs)
+			if err := c.attemptConnections(addrs); err != nil && c.options.onConnectionLostHandler != nil {
+				c.options.onConnectionLostHandler(err)
+			}
 		}
 	}
 }
 
-func (c *Client) attemptConnection(addrs []TCPAddress) {
+func (c *Client) attemptConnections(addrs []TCPAddress) error {
 	if len(addrs) == 0 {
-		return
+		return nil
+	}
+
+	if c.options.multiConnectionMode {
+		return c.attemptMultiConnections(addrs)
+	}
+
+	return c.attemptSingleConnection(addrs)
+}
+
+func (c *Client) attemptMultiConnections(addrs []TCPAddress) error {
+	clients, err := c.multipleClients(addrs)
+	if err != nil {
+		return err
+	}
+
+	if err := c.reloadClients(clients); err != nil {
+		return err
+	}
+
+	return c.resumeSubscriptions()
+}
+
+func (c *Client) resumeSubscriptions() error {
+	c.subMu.RLock()
+	defer c.subMu.RUnlock()
+
+	if len(c.subscriptions) == 0 {
+		return nil
+	}
+
+	if c.options.multiConnectionMode {
+		return c.resumeMultiSubscriptions()
 	}
 
-	// try to start new client first, iff it starts, replace current client
-	cc := c.newClient(addrs, 0)
-	c.reloadClient(cc)
+	// TODO: Handle single connection mode
+	return nil
+}
+
+func (c *Client) resumeMultiSubscriptions() error {
+	// TODO: Handle subscriptions other than shared subscriptions
```

Comment:
> dependent on some non-shared subscription for other use-case where subscribing to all connection will result in duplicate messages?

I am not sure if this will ever exist for backend clients.